bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -0,0 +1,57 @@
+package org.bson.codecs.configuration.mapper;
+
+import com.fasterxml.classmate.MemberResolver;
+import com.fasterxml.classmate.ResolvedType;
+import com.fasterxml.classmate.ResolvedTypeWithMembers;
+import com.fasterxml.classmate.TypeResolver;
+import com.fasterxml.classmate.members.ResolvedField;
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ClassModel extends MappedType {
+    private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
+    private final CodecRegistry registry;
+    private final TypeResolver resolver;
+    
+    public ClassModel(final CodecRegistry registry, final TypeResolver resolver, final Class<?> aClass) {
```

Comment:
I particularly don't like that fasterxml is part of the public API, even in this small way of a constructor parameter